src/main/java/org/lukhnos/nnio/file/Files.java

```diff
@@ -337,14 +337,21 @@ public static long size(Path path) throws IOException {
   }
 
   public static Path walkFileTree(Path start, FileVisitor<? super Path> visitor) throws IOException {
+    File file = start.toFile();
+    if(!file.canRead()){
+      return start;
```

Comment:
Questiont for you: if we really want to implement this method by [the spec](https://docs.oracle.com/javase/tutorial/essential/io/walk.html), should be invoke `visitor.visitFailed(start, someIoException)` here?

Note that this method itself doesn`t even check if pre/post/visit methods want to continue or terminate the walk, and so it doesn't conform to the spec in the first place, but just wanted to check with you.

Reply:
Do you think an AccessDeniedException is adequate here?
visitor.visitFailed(start, new AccessDeniedException(file));

Reply:
> Do you think an AccessDeniedException is adequate here?

Yes, that should do.

Reply:
I added it. Ran it through the java compiler, which complained about file not being converted to string implicitly, so I added a toString() to it.